Expensify/App
# Save message drafts while switching between chats

I was in between issues and didn't understand why this wasn't there so i just added it.

# Fixes
https://github.com/Expensify/ReactNativeChat/issues/445

# Tests
1. Open chat. Type out a message in pinned chat or DM. Dont send the message.
1. Switch to another pinned chat or DM. Type out a message.
1. Switch between the first two chats and confirm the message still remains.
1. Refresh the tab and confirm the message remains.
1. Open the chat in another tab and confirm the draft message remains.
1. Send the message on one of the chat and switch back and forth and confirm the place you sent the message that the draft is cleared.
